src/audio/src/src_ipc4.c

```diff
@@ -107,9 +108,17 @@ int src_set_params(struct processing_module *mod, struct sof_sink *sink)
 	 * as SRC uses period value to calculate its internal buffers,
 	 * it must be done here, right after setting sink parameters
 	 */
-	if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP)
-		dev->period = 1000000 * sink_get_min_free_space(sink) /
+	if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) {
```

Comment:
@jxstelter @singalsu this looks the same as the previous patch in this PR. i.e. we could make this a helper API call to avoid duplication (but can be done as an improvement in another PR).